components/backup-stream/src/subscription_manager.rs

```diff
@@ -0,0 +1,445 @@
+        while let Some(op) = message_box.recv().await {
+            info!("backup stream: on_modify_observe"; "op" => ?op);
+            match op {
+                ObserveOp::Start { region } => {
+                    #[cfg(feature = "failpoints")]
+                    fail::fail_point!("delay_on_start_observe");
+                    self.start_observe(region).await;
+                    IN_FLIGHT_START_OBSERVE_MESSAGE.fetch_sub(1, Ordering::SeqCst);
+                    metrics::INITIAL_SCAN_REASON
+                        .with_label_values(&["leader-changed"])
+                        .inc();
+                }
+                ObserveOp::Stop { ref region } => {
+                    self.subs.deregister_region(region, |_, _| true);
+                }
+                ObserveOp::CheckEpochAndStop { ref region } => {
+                    self.subs.deregister_region(region, |old, new| {
+                        raftstore::store::util::compare_region_epoch(
+                            old.meta.get_region_epoch(),
+                            new,
+                            true,
+                            true,
+                            false,
+                        )
+                        .map_err(|err| warn!("check epoch and stop failed."; "err" => %err))
+                        .is_ok()
+                    });
+                }
+                ObserveOp::RefreshResolver { ref region } => self.refresh_resolver(region).await,
+                ObserveOp::NotifyFailToStartObserve {
+                    region,
+                    handle,
+                    err,
+                } => {
+                    info!("retry observe region"; "region" => %region.get_id(), "err" => %err);
+                    // No need for retrying observe canceled.
+                    if err.error_code() == error_code::backup_stream::OBSERVE_CANCELED {
+                        return;
+                    }
+                    match self.retry_observe(region, handle).await {
+                        Ok(()) => {}
+                        Err(e) => {
+                            self.fatal(
+                                e,
+                                format!("While retring to observe region, origin error is {}", err),
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn fatal(&self, err: Error, message: String) {
+        try_send!(self.scheduler, Task::FatalError(message, Box::new(err)));
+    }
+
+    async fn refresh_resolver(&self, region: &Region) {
+        let need_refresh_all = !self.subs.try_update_region(region);
+
+        if need_refresh_all {
+            let canceled = self.subs.deregister_region(region, |_, _| true);
+            let handle = ObserveHandle::new();
+            if canceled {
+                let for_task = self.find_task_by_region(region).unwrap_or_else(|| {
+                    panic!(
```

Comment:
can we ignore the error and do not panic?

maybe at this time the task is dropped🤔